File: scripts/parse_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');
const path = require('path');

const pdfPath = path.join(__dirname, '../PYQs.pdf');
const outPath = path.join(__dirname, 'raw_pdf.txt');

async function extract() {
  if (!fs.existsSync(pdfPath)) {
    console.error('File not found:', pdfPath);
    return;
  }
  
  const dataBuffer = fs.readFileSync(pdfPath);
  console.log('Extracting text from PDF...');
  
  try {
    const data = await pdf(dataBuffer);
    fs.writeFileSync(outPath, data.text);
    console.log(`Saved ${data.text.length} bytes to raw_pdf.txt`);
  } catch (err) {
    console.error('Error parsing PDF:', err);
  }
}

extract();

File: scripts/parse_pdf.py
import os
from pypdf import PdfReader

script_dir = os.path.dirname(os.path.abspath(__file__))
pdf_path = os.path.join(os.path.dirname(script_dir), 'PYQs.pdf')
out_path = os.path.join(script_dir, 'raw_pdf.txt')

if not os.path.exists(pdf_path):
    print(f"File not found: {pdf_path}")
    exit(1)

text = ""
with open(pdf_path, 'rb') as f:
    reader = PdfReader(f)
    for page in reader.pages:
        page_text = page.extract_text()
        if page_text:
            text += page_text + "\n"

with open(out_path, 'w', encoding='utf-8') as out_f:
    out_f.write(text)

print(f"Saved {len(text)} characters to raw_pdf.txt")
